docs/data/data-grid/list-view/ListViewAdvanced.tsx

```diff
@@ -0,0 +1,368 @@
+      const file = event.target.files[0];
+      const createdAt = new Date().toISOString();
+
+      const fileType = file.type.split('/')[1];
+
+      // validate file type
+      if (!FILE_TYPES.includes(fileType as FileType)) {
+        alert('Invalid file type');
+        return;
+      }
+
+      const row: RowModel = {
+        id: randomId(),
+        name: file.name,
+        description: '',
+        type: fileType as FileType,
+        size: file.size,
+        createdBy: 'Kenan Yusuf',
+        createdAt,
+        updatedAt: createdAt,
+        state: 'pending',
+      };
+
+      event.target.value = '';
+
+      // Add temporary row
+      setLoading(true);
+      setRows((prevRows) => [...prevRows, row]);
+
+      // Simulate server response time
+      const timeout = Math.floor(Math.random() * 3000) + 2000;
+      setTimeout(() => {
+        const uploadedRow: RowModel = { ...row, state: 'uploaded' };
+        setRows((prevRows) =>
+          prevRows.map((r) => (r.id === row.id ? uploadedRow : r)),
+        );
+        setDetailsState({ open: true, params: { row } });
+        setLoading(false);
+      }, timeout);
+    },
+    [],
+  );
+
+  const columns: GridColDef[] = React.useMemo(
+    () => [
+      {
+        field: 'name',
+        headerName: 'Name',
+        width: 350,
+        editable: true,
+        hideable: false,
+        renderCell: (params) => {
+          return (
+            <Stack
+              direction="row"
+              gap={1.5}
+              alignItems="center"
+              sx={{ height: '100%' }}
+            >
+              <FileIcon type={params.row.type} />
+              {params.value}
+            </Stack>
+          );
+        },
+      },
+      {
+        field: 'createdBy',
+        headerName: 'Owner',
+        width: 200,
+        renderCell: (params) => {
+          const avatarProps = stringAvatar(params.value);
+          return (
+            <Stack direction="row" gap={1.5} alignItems="center">
+              <Avatar
+                {...avatarProps}
+                sx={{ width: 24, height: 24, fontSize: 12, ...avatarProps.sx }}
+              />
+              {params.value}
+            </Stack>
+          );
+        },
+      },
+      {
+        field: 'createdAt',
+        headerName: 'Added',
+        type: 'date',
+        width: 200,
+        valueFormatter: formatDate,
+      },
+      {
+        field: 'updatedAt',
+        headerName: 'Modified',
+        type: 'date',
+        width: 200,
+        valueFormatter: formatDate,
+      },
+      {
+        field: 'type',
+        headerName: 'Type',
+        width: 150,
+      },
+      {
+        field: 'size',
+        headerName: 'Size',
+        width: 120,
+        valueFormatter: formatSize,
+      },
+      {
+        type: 'actions',
+        field: 'actions',
+        resizable: false,
+        width: 50,
+        getActions: (params) =>
+          isListView
+            ? [
+                <ActionDrawer
```

Comment:
Nitpick: it would be better to lift this drawer up and reuse it instead of having it in each row.
This can be done in a follow-up though.

Reply:
Made that change here https://github.com/mui/mui-x/pull/14393/commits/9e5f8b3ca2715ca7d20d156176eae2ee3be05dbc